modelci/types/models/mlmodel.py

```diff
@@ -165,3 +165,21 @@ def check_model_info(cls, values: dict):  # pylint: disable=no-self-use
     def saved_path(self):
         suffix = Path(self.weight).suffix
         return super().saved_path.with_suffix(suffix)
+
+
+class ModelUpdate(BaseModel):
+    architecture: Optional[str] = Field(None, example='ResNet50')
+    framework: Optional[Framework]
+    engine: Optional[Engine]
+    version: Optional[PositiveInt] = Field(None, example=1)
+    dataset: Optional[str] = Field(None, example='ImageNet')
+    metric: Optional[Dict[Metric, float]] = Field(None, example='{"acc": 0.76}')
+    task: Optional[Task]
+    inputs: Optional[List[IOShape]] = Field(
+        [],
```

Comment:
better to use `default_factory` so that the list is dynamically generated only when the data object is created.